lib/amplitude.dart

```diff
@@ -1,342 +1,211 @@
+  /// Set a custom user Id.
   ///
-  /// Note: This operation is irreversible!!
-  Future<void> clearUserProperties() async {
-    return await _channel.invokeMethod(
-        'clearUserProperties', jsonEncode(_baseProperties()));
-  }
-
-  /// Upload all unsent events.
-  Future<void> uploadEvents() async {
-    return await _channel.invokeMethod(
-        'uploadEvents', jsonEncode(_baseProperties()));
-  }
-
-  Map<String, dynamic> _baseProperties() {
-    return {'instanceName': _instanceName};
-  }
-
-  /// Private bridging calls
-  Future<void> _setLibraryName(String libraryName) async {
-    Map<String, dynamic> properties = _baseProperties();
-    properties['libraryName'] = libraryName;
-
-    return await _channel.invokeMethod(
-        'setLibraryName', jsonEncode(properties));
-  }
-
-  Future<void> _setLibraryVersion(String libraryVersion) async {
-    Map<String, dynamic> properties = _baseProperties();
-    properties['libraryVersion'] = libraryVersion;
-
-    return await _channel.invokeMethod(
-        'setLibraryVersion', jsonEncode(properties));
-  }
-
-  /// Fetches the deviceId, a unique identifier shared between multiple users using the same app on the same device.
-  /// @returns the deviceId.
-  Future<String?> getDeviceId() async {
-    return await _channel.invokeMethod(
-        'getDeviceId', jsonEncode(_baseProperties()));
-  }
-
-  /// Fetches the userId, a unique identifier for tracking a user.
-  /// @returns the userId
-  Future<String?> getUserId() async {
-    return await _channel.invokeMethod(
-        'getUserId', jsonEncode(_baseProperties()));
-  }
-
-  /// Fetches the current sessionId, an identifier used by Amplitude to group together events tracked during the same session.
-  /// @returns the sessionId.
-  Future<int?> getSessionId() async {
-    return await _channel.invokeMethod(
-        'getSessionId', jsonEncode(_baseProperties()));
-  }
+  /// If your app has its own login system that you want to track users with,
+  /// you can set the userId.
+  Future<void> setUserId(String? userId) async {
+    Map<String, dynamic> properties = {};
+    properties["setUserId"] = userId;
 
-  // Android only, supports to use android app set id as device id
-  Future<void> useAppSetIdForDeviceId() async {
-    return await _channel.invokeMethod(
-        'useAppSetIdForDeviceId', jsonEncode(_baseProperties()));
+    return await _channel.invokeMethod("setUserId", jsonEncode(properties));
   }
 
-  /// Defines a custom session expiration time where the timeout input is in milliseconds.
-  Future<void> setMinTimeBetweenSessionsMillis(int timeInMillis) async {
-    Map<String, dynamic> properties = _baseProperties();
-    properties['timeInMillis'] = timeInMillis;
+  /// Sets a custom device ID.
+  ///
+  /// Make sure the value is sufficiently unique. Amplitude recommends using a UUID.
+  Future<void> setDeviceId(String deviceId) async {
+    Map<String, dynamic> properties = {};
```

Comment:
Is this needed?

Reply:
I'd like to follow the pattern and keep the argument in `invokeMethod<T>(String method, [dynamic arguments])` to be a json string so that all messages can be decoded on clients regardless of `method`.